src/OrchardCore.Modules/OrchardCore.Rules/Drivers/ContentTypeConditionEvaluatorDriver.cs

```diff
@@ -43,6 +44,14 @@ public ValueTask<bool> EvaluateAsync(Condition condition)
         private ValueTask<bool> EvaluateAsync(ContentTypeCondition condition)
         {
             var operatorComparer = _operatorResolver.GetOperatorComparer(condition.Operation);
+
+            // If no content types are considered and the operation is an INegateOperator, return true,
+            // since displaying no content type should match any negative comparision on any content type
+            if (!_contentTypes.Any() && condition.Operation is INegateOperator)
+            {
+                return ValueTask.FromResult(true);
+            }
```

Comment:
I'm not sure we can say that this will be correct for all `INegateOperator` implementations (the current built-in ones, yes, but this isn't necessarily future-proof and appropriate for third-party implementations). How about running `operatorComparer` with an empty string instead?

Reply:
I had thought about that. I was originally thinking I only wanted to do that for negative comparisons, but you're right, that should return the expected result (false) for the other comparisons as well. I'll go ahead and make that change.